kbase/narrative
# DATAUP-310 state management: adding app specs and toggling between them

# Description of PR purpose/changes
Let's try this again.

This has a few updates in support of bulk import cell state management. Much of this is setup for handling the state that the cell should be in, and much of that is based on what the user's inputs are, which in turn depends on what the apps are that need to be run. So this does a few things to set that up.

**1. Adds fetching of app specs at cell startup.** 
Now, when starting a bulk import cell, this fetches the app specs from the Narrative Method Store service. Fetching these means changing around the cell setup and initialization code. It also shifts around the inputs to the cell to look like this:
```
{
    importData: {
        fileType: {
            files: ['file1'],
            appId: 'SomeModule/someapp'
        },
        specs: {
            'SomeModule/someapp': { complicated app spec }
        }
    }
}
```
I added this structure, instead of just adding the spec into the fileType, as multiple file types can use the same spec. It seemed cleaner. They get stored in the cell metadata this way, too.

This kinda opened a rabbit hole that we need to go down anyway, which is adapting the `model` object to be useful and directly represent what we're storing in the metadata. Other widgets were modified to handle that.

**2. Adds a first step at the state management.**
This will, in a subsequent PR, function as the FSM does for the app cell. Right now there's just a single state - "new" - that is set for configuration.

**3. Implements the file type toggle.**
Now that there's different specs available for different apps, changing the active file type should also change the active configuration tab to reflect the app spec. This means having the configuration tab (and filePathWidget and paramsWidget) aware of the currently active spec.

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-X

# Testing Instructions
* Details for how to test the PR: 
You'll need to create a new Bulk Import cell locally for this to operate right. You'll probably see some strange errors otherwise. Just make a new cell with the usual file selection methods.

# Dev Checklist:
